app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_settings_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path, schema = 'http')
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_host(@host, content_path)
+      return content_path if content_path.start_with?('http', 'https', 'ftp', 'ftps')
+      return "#{foreman_settings_url}/pulp/repos/#{path}" unless @host.content_source
```

Comment:
When testing with actually uploading the squashfs file to katello, I've noticed, that it was placed under `/pulp/isos`, e.g.:

```
http://sat63-qa-rhel7.example.com/pulp/isos/Default_Organization/Library/custom/RHV/rhvh/squashfs.img
```

I guess the `repository_url` should take also this info as argument, something like:

```
def repository_url(content_path, content_type = 'repos', schema = 'http')
```

Reply:
I'm not sure that I understand where the `content_type` should be placed.

Reply:
`"#{foreman_settings_url}/pulp/#{contetn_type}/#{path}"`